src/mscorlib/shared/System/TimeZoneInfo.Win32.cs

```diff
@@ -37,9 +37,21 @@ private sealed partial class CachedData
             private static TimeZoneInfo GetCurrentOneYearLocal()
             {
                 // load the data from the OS
-                Win32Native.TimeZoneInformation timeZoneInformation;
-                long result = UnsafeNativeMethods.GetTimeZoneInformation(out timeZoneInformation);
-                return result == Win32Native.TIME_ZONE_ID_INVALID ?
+                long result;
+                Interop.Kernel32.TIME_ZONE_INFORMATION timeZoneInformation;
+
+#if CORERT
+                Interop.Kernel32.TIME_DYNAMIC_ZONE_INFORMATION dynamicTimeZoneInformation;
+                result = Interop.Kernel32.GetDynamicTimeZoneInformation(out dynamicTimeZoneInformation);
```

Comment:
We can use GetDynamicTimeZoneInformation in both cases. the coreclr implementation using GetTimeZoneInformation because the code picked up from the desktop implementation which needed to run on the downlevel OS. 
GetDynamicTimeZoneInformation exists since Vista so we can use it on coreclr too as the minimum support is Windows 7.